stups-spring-oauth2-server/src/main/java/org/zalando/stups/oauth2/spring/server/AbstractAuthenticationExtractor.java

```diff
@@ -26,59 +26,75 @@
 import org.springframework.security.oauth2.common.exceptions.InvalidTokenException;
 import org.springframework.security.oauth2.provider.OAuth2Authentication;
 import org.springframework.security.oauth2.provider.OAuth2Request;
+
 import org.springframework.util.StringUtils;
 
+import org.zalando.stups.oauth2.spring.authorization.UserRolesProvider;
+
 /**
  * Common code for current {@link AuthenticationExtractor}-implementations.
  *
- * @author jbellmann
+ * @author  jbellmann
  */
 public abstract class AbstractAuthenticationExtractor implements AuthenticationExtractor {
 
-    private static final String UID_SCOPE = "uid";
+    private static final String UID = "uid";
+
+    public static final String ACCESS_TOKEN = "access_token";
+
+    public static final String REALM = "realm";
+
+
 
     private boolean throwExceptionOnEmptyUid = true;
 
     @Override
-    public OAuth2Authentication extractAuthentication(final Map<String, Object> tokenInfoMap, final String clientId) {
-        UsernamePasswordAuthenticationToken user = createAuthenticationToken(tokenInfoMap);
+    public OAuth2Authentication extractAuthentication(final Map<String, Object> tokenInfoMap, final String clientId,
+            final UserRolesProvider userRolesProvider) {
+        final UsernamePasswordAuthenticationToken user = createAuthenticationToken(tokenInfoMap, userRolesProvider);
 
         // at the moment there is other way
-        Set<String> scopes = resolveScopes(tokenInfoMap);
+        final Set<String> scopes = resolveScopes(tokenInfoMap);
 
         return buildOAuth2Authentication(user, scopes, clientId);
     }
 
-    protected UsernamePasswordAuthenticationToken createAuthenticationToken(final Map<String, Object> tokenInfoMap) {
-        List<GrantedAuthority> authorities = createAuthorityList(tokenInfoMap);
+    protected UsernamePasswordAuthenticationToken createAuthenticationToken(final Map<String, Object> tokenInfoMap,
```

Comment:
Generally a lot of small changes that make it really hard to know what the "real" changes are... I heard it makes unicorns die... sometimes...